open-craft/ansible-playbooks
# [BB-2836] Monitoring grade report generation - Setup security features for elasticsearch and kibana

This PR enables security features for elasticsearch which is required to use kibana alert system.
Following new variables are introduced, which I think should be set from configuration secure repository -

- ``elasticsearch_password`` - Password that will be used for ``elastic`` and ``kibana_system`` user in elastic.
- ``kibana_elasticsearch_password`` - Password for ``kibana_system`` user to connect with elasticsearch. Essentially same as ``elasticsearch_password``.
- ``kibana_encryption_key`` - Kibana uses for encrypting and decrypting sensitive information.
- ``elasticsearch_pem_file`` - Elastic search CA pem file location for kibana.

This PR also introduces a new role - ``kibana-email-alert``. Which enables a ``systemd`` service that watches over Kibana logs on ``journald`` with a predefined regex. If there a Kibana Alert with ``Server Log`` connector and that Alert triggers, this new service collects that log and notifies to a given email address. New variables for this role -

- ``kibana_email_alert_regex`` - what to match in journald log, defaults to ``Server log``
- ``kibana_email_alert_to``- which email to send alert, **Required**
- ``kibana_email_alert_subject`` - email subject, defaults to "Kibana Error Log"

**JIRA tickets**: https://tasks.opencraft.com/browse/BB-2836

~~**Discussions**:~~

**Dependencies**: None

~~**Screenshots**:~~

~~**Sandbox URL**:~~

**Merge deadline**: "None" 

**Testing instructions**:

1. Deploying Kibana should now setup elasticsearch bootstrap password and ``kibana_system`` user password
2. Kibana should be able to connect to elasticsearch using that password
3. Alerting feature is enabled in Kibana
4. Create a ``Server Log`` connector
5. Test that connector if email alert working
6. Create a Kibana Alert & set the ``Server Log`` connector as action
7. When Alert triggers, an email should go to the address defined in ``kibana_email_alert_to`` variable

**Author notes and concerns**:
1. Email contents are not properly formatted
2. Is there a better regex other than ``Server log``?

**Reviewers**

~~**Settings**~~
